### ISSUE TITLE
[Bug] [Python] Use Python report permission error

### ISSUE THREAD

**Issue by stalary** [apache/dolphinscheduler#10533]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

[INFO] 2022-06-21 16:15:17.133 +0800 [taskAppId=TASK-20220621-5935139969120_4-21-24] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.python.PythonTask:[63] -  -> /tmp/dolphinscheduler/exec/process/5934261425632/5935139969120_4/21/24/21_24.command:行5: /tmp/dolphinscheduler/exec/process/5934261425632/5935139969120_4/21/24/py_21_24.py: 权限不够

### What you expected to happen

success exec python.

### How to reproduce

use 3.0.0-beat1, create tenant, use it to exec python.

### Version

3.0.0-beta-1

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by stalary** [comment]
I use pseudo-cluster.

**Comment by stalary** [comment]
I have located the problem. I have customized the environment, which causes the global environment of DS to fail and Python Home can not be found.

**Comment by SbloodyS** [comment]
Hi @stalary , would you like to submit a PR to fix this?

**Comment by stalary** [comment]
> Hi @stalary , would you like to submit a PR to fix this?

I can fix it, but first we need to talk about whether ideally it should be the highest priority task usage environment, the second priority DS environment, and merge at the same time.

**Comment by SbloodyS** [comment]
Sorry. I took a closer look at this issue. It seems the user you started ds did not have the `sudo` permission of execute the shell script. Not bug.

**Comment by stalary** [comment]
> Sorry. I took a closer look at this issue. It seems the user you started ds did not have the `sudo` permission of execute the shell script. Not bug.

I think this is an environment related issue, the error is that the Python environment variable was not found, and this variable was configured in the startup DS configuration

**Comment by SbloodyS** [comment]
> I think this is an environment related issue, the error is that the Python environment variable was not found, and this variable was configured in the startup DS configuration

Yes. And the `PYTHON_HOME` variable can be customized in `https://dolphinscheduler.apache.org/en-us/docs/latest/user_doc/guide/security.html Environmental Management`. You can try this.

**Comment by stalary** [comment]
> > I think this is an environment related issue, the error is that the Python environment variable was not found, and this variable was configured in the startup DS configuration
> 
> Yes. And the `PYTHON_HOME` variable can be customized in `https://dolphinscheduler.apache.org/en-us/docs/latest/user_doc/guide/security.html Environmental Management`. You can try this.

Yes, I solved the problem with this, but why didn't the parameters I configured at startup take effect. 
export PYTHON_HOME=/opt/anaconda3/bin/python3.9
I think normal logic would have worked.

**Comment by SbloodyS** [comment]
Did you define `PYTHON_HOME` in `bin/env/dolphinscheduler_env.sh` and start the service by `bin/dolphinscheduler-daemon.sh start xxx-server`? @stalary

**Comment by stalary** [comment]
> Did you define `PYTHON_HOME` in `bin/env/dolphinscheduler_env.sh` and start the service by `bin/dolphinscheduler-daemon.sh start xxx-server`? @stalary

I define it and use bin/start-all.sh.

**Comment by SbloodyS** [comment]
Could you provide more detailed information since i can not reproduce this problem in my local environment? @stalary

**Comment by stalary** [comment]
> Could you provide more detailed information since i can not reproduce this problem in my local environment? @stalary

first create a environment file，write other export command，not python_home.
second，use it execute task，the startup python_home will be lose.

**Comment by SbloodyS** [comment]
> > Could you provide more detailed information since i can not reproduce this problem in my local environment? @stalary
> 
> first create a environment file，write other export command，not python_home. second，use it execute task，the startup python_home will be lose.

If you want to perform Python tasks, then `python_home` is required.

**Comment by stalary** [comment]
> > > Could you provide more detailed information since i can not reproduce this problem in my local environment? @stalary
> > 
> > 
> > first create a environment file，write other export command，not python_home. second，use it execute task，the startup python_home will be lose.
> 
> If you want to perform Python tasks, then `python_home` is required.

I think PYTHON_HOME configured at startup should be merge into the task environment.

**Comment by SbloodyS** [comment]
> first create a environment file，write other export command，not python_home. second，use it execute task，the startup python_home will be lose.

But you did not export `PYTHON_HOME`.

**Comment by zhongjiajie** [comment]
FYI, see https://github.com/apache/dolphinscheduler/issues/10562#issuecomment-1175877731, I think we could improve it

**Comment by zzzhangqi** [comment]
I had the same problem. 3.0.0-beta-2 run in docker.
set docker env `PYTHON_HOME=/usr/bin/python3`, running python task does not take effect,Prompt does not have permission.

Successful operation:
set env `PYTHON_HOME=/usr/bin/python3` in environment-manage,python task works normally.

Why didn't `/opt/dolphinscheduler/conf/dolphinscheduler_env.sh` take effect?

**Comment by haozileung** [comment]
Same issue here.

After debug in worker node, it turns out that

`sudo -u root -i /tmp/dolphinscheduler/exec/process/default/11497277068832/11497401001760_3/11/11/11_11.sh`

the sudo command did not perserved the env from original shell's configuration.

The workaround would be define an environment manually and specified when running the tasks.

**Comment by hieunm89** [comment]
Having the same issue, I have export PYTHON_LAUNCHER=${PYTHON_LAUNCHER:-/usr/bin/python3} set in dolphinscheduler_env.sh  as well as "export PYTHON_HOME=/usr/bin/python3" in Environment Config. Nothing works

**Comment by XiaoZhiSnow** [comment]
export PYTHON_LAUNCHER=${PYTHON_LAUNCHER:-/usr/bin/python}
export PATH=$HADOOP_HOME/bin:$SPARK_HOME2/bin:$PYTHON_LAUNCHER:$JAVA_HOME/bin:$HIVE_HOME/bin:$FLINK_HOME/bin:$DATAX_HOME/bin:$SEATUNNEL_HOME/bin:$CHUNJUN_HOME/bin:$PYTHON_LAUNCHER:$PATH

but it does not work at 3.2.0

**Comment by XiaoZhiSnow** [comment]
config PYTHON_LAUNCHER at /etc/profile  
source /etc/profile
restart ds, then it will work

**Comment by cnfree0355** [comment]
**I have the same issue with version 3.2** 
Here is my environment setup in dolphinscheduler_env.sh

```bash
  export SPARK_HOME=${SPARK_HOME:-/data/spark3}

  export PYTHON_LAUNCHER=${PYTHON_LAUNCHER:-/usr/local/python3/bin/python3}

  export DATAX_LAUNCHER=${DATAX_LAUNCHER:-/usr/local/datax/bin/datax.py}

  export PATH=$HADOOP_HOME/bin:$SPARK_HOME/bin:$PYTHON_LAUNCHER:$JAVA_HOME/bin:$HIVE_HOME/bin:$FLINK_HOME/bin:$DATAX_LAUNCHER:$PATH
```


When I pass the task or the result of these variables

```bash
echo " ------shell task 1 ------"
echo "The python launcher program is ${PYTHON_LAUNCHER} "
echo "The spark home is ${SPARK_HOME}"
echo "The datax launcher program is ${DATAX_LAUNCHER}"`
```
As shown in the log below, they are all null.

```text
 ------shell task 1 ------
	The python launcher program is  
	The spark home is 
	The datax launcher program is 
```

**Comment by niyanchun** [comment]
same problem in 3.2.1 version, any idea?

**Comment by tpang-cxl** [comment]
same problem in 3.2.1 version as well

**Comment by jmorcar** [comment]
The problem is solved in 3.2.2 using Security > Environment Manage

<img width="1754" alt="Image" src="https://github.com/user-attachments/assets/07697487-b327-48b4-a08b-c461cc3f6bea" />

After in your Workflow definition, define Start execution of workflow indicating the specific environment item created before:

 
<img width="585" alt="Image" src="https://github.com/user-attachments/assets/d5d87be4-cf7b-4bdc-be1d-d580deac1729" />
